src/system/SystemMutex.cpp

```diff
@@ -110,6 +110,42 @@ DLL_EXPORT void Mutex::Lock(void)
 }
 #endif // CHIP_SYSTEM_CONFIG_FREERTOS_LOCKING
 
+#if CHIP_SYSTEM_CONFIG_PLATFORM_LOCKING
+DLL_EXPORT CHIP_ERROR Mutex::Init(Mutex & aThis)
+{
+restart:
+    if (__sync_bool_compare_and_swap(&aThis.mInitialized, 0, 1))
+    {
+        platform_init_mutex((platform_mutex_t *) &aThis.mSemaphore);
```

Comment:
Please use static_cast or reinterpret_cast, not C-style casts.  But why is this cast needed at all?